simulator/infra/replacement/cache_replacement.cpp

```diff
@@ -84,10 +85,110 @@ void LRUCacheInfo::erase_lru_element() {
     lru_hash.erase( lru_elem);
 }
 
+class Pseudo_LRUCacheInfo : public CacheReplacementInterface
+{
+    public:
+        explicit Pseudo_LRUCacheInfo( std::size_t ways);
+        void touch( std::size_t way) override;
+        std::size_t update() override;
+        std::size_t get_ways() const override { return ways; }
+
+        void set_to_erase( std::size_t ) override;
+        void allocate( std::size_t ) override;
+
+    private:
+        enum flags { LEFT, RIGHT};
```

Comment:
Please follow our naming conventions. Types should start with capital letter (`Flag`).